extensions/wikia/CommunityPage/models/CommunityPageSpecialInsightsModel.class.php

```diff
@@ -72,29 +85,32 @@ private function getInsightModule( $type, $sortingType = self::INSIGHTS_MODULE_S
 
 		if ( $insightPages['count'] > self::INSIGHTS_MODULE_ITEMS ) {
 			$insightPages['fulllistlink'] = SpecialPage::getTitleFor( 'Insights', $type )
-				->getLocalURL( $this->getSortingParam( $sortingType ) );
+				->getLocalURL( $this->getSortingParam( $config['sortingType'] ) );
 		}
 
-		$insightPages = $this->addLastRevision( $insightPages );
+		$insightPages = $this->addLastRevision( $insightPages, $config['displayPageviews'] );
 
 		return $insightPages;
 	}
 
 	/**
 	 * @param array $insightsPages
+	 * @param boolean $displayPageviews should display information about pageviews
 	 * @return array Prepare message about who and when last edited given article
 	 * @throws MWException
 	 */
-	private function addLastRevision( $insightsPages ) {
+	private function addLastRevision( $insightsPages, $displayPageviews ) {
+		global $wgLang;
+
 		foreach ( $insightsPages['pages'] as $key => $insight ) {
 			$insightsPages['pages'][$key]['metadataDetails'] = $this->getArticleMetadataDetails( $insight['metadata'] );
 			$insightsPages['pages'][$key]['editlink'] = $this->getEditUrl( $insight['link']['articleurl'] );
 			$insightsPages['pages'][$key]['edittext'] = $this->getArticleContributeText( $insight['metadata'] );
 
-			if ( !empty( $insight['metadata']['pv7'] ) ) {
+			if ( $displayPageviews ) {
```

Comment:
What if `$insight['metadata']['pv7'] ` will be empty? Sounds like potential place for throwing an error?
Simple Assert::true( !empty( $insight['metadata']['pv7'] ), 'log message'  ) would let us be notified by ErrorReported if there are many of those.